### PR TITLE
chore(argo-rollouts): Upgrade argo-rollouts to 1.3.1

### PR DESCRIPTION
**Pull request by jmeridth**
[Release notes](https://github.com/argoproj/argo-rollouts/releases/tag/v1.3.1)

Signed-off-by: jmeridth <jmeridth@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-rollouts/README.md.gotmpl [hunk 1]
```diff
@@ -78,10 +78,10 @@ If dashboard is installed by `--set dashboard.enabled=true`, checkout the argo-r
 
 ### To 1.0.0
 
-* This is a breaking change which only supports Helm v3.0.0+ now. If you still use Helm v2, please consider upgrading because v2 is EOL since November 2020.   
+* This is a breaking change which only supports Helm v3.0.0+ now. If you still use Helm v2, please consider upgrading because v2 is EOL since November 2020.
```
Comment (mkilchhofer) [inline charts/argo-rollouts/README.md.gotmpl:81]: Why this change? I assume that 2 trailing spaces are markdown newlines

File: charts/argo-rollouts/README.md.gotmpl [hunk 2]
```diff
@@ -78,10 +78,10 @@ If dashboard is installed by `--set dashboard.enabled=true`, checkout the argo-r
...
   To migrate to Helm v3 please have a look at the [Helm 2to3 Plugin](https://github.com/helm/helm-2to3). This tool will convert the existing ConfigMap used for Tiller to a Secret of type `helm.sh/release.v1`.
 * `quay.io` is the default registry now
-* We introduce a template function for the labels here to reduce code duplication. This also affects the Deployment `matchLabels` selector.
```
Comment (mkilchhofer) [inline charts/argo-rollouts/README.md.gotmpl:84]: Same here

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-rollouts/README.md.gotmpl (hunk 1 above)
Comment (jmeridth) [reply charts/argo-rollouts/README.md.gotmpl:81]: Nope.  Just spaces.  My local editor keeps auto-trimming on save so my commit gets out of whack with generated.  Hence the removal from `README.md.gotmpl`.  Put back still or can remove.

I am also tempted to remove the last line from the README so I don't have to include a README edit every time I update the version of this chart.  Not a fan.  wdyt?

Comment (mkilchhofer) [reply charts/argo-rollouts/README.md.gotmpl:81]: Okay 👍 i think we should introduce a .editorconfig in the repos root dir ;)

Yup, you could replace it with `main` instead of the version. As a downside users are always redirected to the newest values.yaml file. But thats how all other charts are working.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[*.md]
```
Comment (mkilchhofer) [inline .editorconfig:1]:
```suggestion
[*.{md,md.gotmpl}]
```

File: charts/argo-rollouts/README.md.gotmpl (hunk 1 above)
Comment (mkilchhofer) [reply charts/argo-rollouts/README.md.gotmpl:81]: Btw: it looks different ;) 
grafik

Comment (jmeridth) [reply charts/argo-rollouts/README.md.gotmpl:81]: Going to put the spaces back in.  I'm also getting a markdown preview plugin for vscode 😄

Comment (mkilchhofer) [reply charts/argo-rollouts/README.md.gotmpl:81]: 🤣🤣

Comment (jmeridth) [reply charts/argo-rollouts/README.md.gotmpl]: spaces back, editorconfig being respected.  awwwwyeah

**Comment by jmeridth** [comment]
@mkilchhofer this is good to go I think.

**Review by mkilchhofer — APPROVED** [review]
